src/components/tables/plainTable/chackraPlainTable.tsx

```diff
@@ -1,73 +1,85 @@
-import {
-  Table,
-  Thead,
-  Tbody,
-  Tr,
-  Th,
-  Td,
-  TableContainer,
-  Center,
-  Spinner,
-} from '@chakra-ui/react'
+import { Center, Spinner, Table, TableContainer, Tbody, Td, Th, Thead, Tr } from "@chakra-ui/react";
+import type { SerializedError } from "@reduxjs/toolkit";
+import type { FetchBaseQueryError } from "@reduxjs/toolkit/query";
+import type { z, ZodObject, ZodRawShape } from "zod";
 
-export type PlainTablePropsType<T> = {
-  data: T[]
-  colorscheme?: string
-  variant?: string
-  isLoading: boolean
-  isError: boolean
-}
+import type { DetailsType } from "../../../lib/errorHandler/errorHandler";
+import { setError } from "../../../lib/errorHandler/errorHandler";
+import { isErrorWithMessage, isFetchBaseQueryError } from "../../../lib/errorHandler/errorHandlingLib";
 
-export const ChackraPlainTable = <T extends object,>({
+export type PlainTablePropsType<T extends ZodRawShape> = {
+  data: z.infer<ZodObject<T>>[];
+  colorscheme?: string;
+  variant?: string;
+  isLoading: boolean;
+  isError: boolean;
+  schema: ZodObject<T>;
+  error: FetchBaseQueryError | SerializedError | undefined;
+};
+
+export const ChackraPlainTable = <T extends ZodRawShape>({
   data,
-  colorscheme = 'grey',
-  variant = 'striped',
+  colorscheme = "grey",
+  variant = "striped",
   isLoading,
   isError,
+  error,
+  schema,
 }: PlainTablePropsType<T>) => {
-  type TK = keyof T
-  const first = data.length < 1 ? {} : data[0];
-
-  /* TODO: this doesn't work well as
-   * Typescript doesn't support compile-time to derive the properties of T
-   * At runtime, the Array may be empty and thus cannot be derived either
-   * Solutions:
-   *   1. move responsability to developer
-   *   2. use Zod schema, pairing T with Schema: z.object({...}).shape contains the list of keys
-   *      requires time to understand the right Generics to use 
-   */
-  const headers = Object.keys(first) as TK[]
-
-  return (
-    <TableContainer my="30px">
-      <Table variant={variant} colorScheme={colorscheme}>
-        <Thead>
-          <Tr>
-            {headers.map((header) => (
-              <Th key={String(header)}>{String(header)}</Th>
-            ))}
-          </Tr>
-        </Thead>
-        <Tbody>
-          {isLoading || isError ? (
+  if (isError && error) {
+    let details = {} as DetailsType;
+    if (isFetchBaseQueryError(error)) {
+      details = {
+        title: "Fetching Error",
+        message: "error" in error ? error.error : JSON.stringify(error.data),
+        status: error.status,
+        isValidationError: false,
+        exceptionDetails: null,
+      } as DetailsType;
+    } else if (isErrorWithMessage(error)) {
+      // you can access a string 'message' property here
+      details = {
+        title: "Fetching Error",
+        message: error.message,
+        isValidationError: false,
+        exceptionDetails: null,
+      } as DetailsType;
+    }
+    setError({ error: true, details });
+    return <></>;
```

Comment:
I'm not sure setting the global error is the right approach and not to display the Table, thus changing a lot the page display.

1. Ensure that on fetch error, the display below the error popup is ok
2. Should in this case the error have a 'reload' or else, after an error, how the page can recover without a full SPA reload!
3. Errors in background polling (refresh) not sure should break the flow. The `isLoading` fair currently doesn't toggle on background polling (`isFetching` is for it) and the spinner do not replace the table during polling. Same for errors. Afair you can have both {data} and {error} in that case. Please check rtkq docs.

Reply:
Now the table is still visible and if an error occour after the first fetch (while polling), in the example, we show a toast.
Added a comment to it also.
Checking isFetching other than isLoading is page duty to me so I added it there.
Added a reload page button on the error modal